File: character-copy/csharp/src/CharacterCopy/CharacterCopy.cs
namespace CharacterCopy;

public static class CharacterCopy
{
    public static void Copy(ISource source, IDestination destination)
    {
        while (true)
        {
            char c = source.ReadChar();
            if (c == '\n') return;
            destination.WriteChar(c);
        }
    }
}

File: character-copy/typescript/src/characterCopy.ts
import type { Source } from './Source.js';
import type { Destination } from './Destination.js';

export function copy(source: Source, destination: Destination): void {
  while (true) {
    const ch = source.read();
    if (ch === '\n') return;
    destination.write(ch);
  }
}
